langchain4j-bedrock/src/main/java/dev/langchain4j/model/bedrock/converse/BedrockChatModel.java

```diff
@@ -0,0 +1,346 @@
+import software.amazon.awssdk.services.bedrockruntime.model.InferenceConfiguration;
+import software.amazon.awssdk.services.bedrockruntime.model.Message;
+import software.amazon.awssdk.services.bedrockruntime.model.SpecificToolChoice;
+import software.amazon.awssdk.services.bedrockruntime.model.StopReason;
+import software.amazon.awssdk.services.bedrockruntime.model.SystemContentBlock;
+import software.amazon.awssdk.services.bedrockruntime.model.Tool;
+import software.amazon.awssdk.services.bedrockruntime.model.ToolChoice;
+import software.amazon.awssdk.services.bedrockruntime.model.ToolConfiguration;
+import software.amazon.awssdk.services.bedrockruntime.model.ToolInputSchema;
+import software.amazon.awssdk.services.bedrockruntime.model.ToolResultBlock;
+import software.amazon.awssdk.services.bedrockruntime.model.ToolResultContentBlock;
+import software.amazon.awssdk.services.bedrockruntime.model.ToolResultStatus;
+
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+import static dev.langchain4j.internal.RetryUtils.withRetry;
+import static java.util.Collections.emptyList;
+import static java.util.Collections.singletonList;
+import static java.util.stream.Collectors.toList;
+
+public class BedrockChatModel implements ChatLanguageModel {
+
+    public static final String DEFAULT_MODEL_ID = "anthropic.claude-3-5-sonnet-20240620-v1:0";
+
+    private final ObjectMapper objectMapper = new ObjectMapper()
+            .disable(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES)
+            .setSerializationInclusion(JsonInclude.Include.NON_NULL);
+
+    private Region region;
+    private AwsCredentialsProvider credentialsProvider;
+    private String modelId;
+    private InferenceConfiguration inferenceConfiguration;
+    private Integer maxRetries;
+    private Duration timeout;
+    private BedrockRuntimeClient client;
+
+    public BedrockChatModel() {
+        this.region = Region.US_EAST_1;
+        this.credentialsProvider = DefaultCredentialsProvider.builder().build();
+        this.modelId = DEFAULT_MODEL_ID;
+        this.inferenceConfiguration = InferenceConfiguration.builder().build();
+        this.maxRetries = 5;
+        this.timeout = Duration.ofMinutes(1L);
+        this.client = createClient();
+    }
+
+    public BedrockChatModel(
+            Region region,
+            AwsCredentialsProvider credentialsProvider,
+            String modelId,
+            InferenceConfiguration inferenceConfiguration,
+            Integer maxRetries,
+            Duration timeout
+    ) {
+        this.region = region;
+        this.credentialsProvider = credentialsProvider;
+        this.modelId = modelId;
+        this.inferenceConfiguration = inferenceConfiguration;
+        this.maxRetries = maxRetries;
+        this.timeout = timeout;
+        this.client = createClient();
+    }
+
```

Comment:
Could you please implement `ChatResponse chat(ChatRequest request)` method as well?